core/lib/zksync_core/src/api_server/tx_sender/mod.rs

```diff
@@ -184,6 +185,7 @@ impl TxSenderBuilder {
         vm_concurrency_limiter: Arc<VmConcurrencyLimiter>,
         api_contracts: ApiContracts,
         storage_caches: PostgresStorageCaches,
+        whitelisted_tokens_for_aa_cache: Arc<RwLock<Vec<Address>>>,
```

Comment:
I guess from the perspective of the node framework, we'd want an enterprise-y interface like

```rust
#[async_trait]
pub trait WhitelistedTokensProvider: 'static + Send + Sync + fmt::Debug {
    async fn whitelisted_tokens(&self) -> Vec<Address>;
}

// Implementation used by the main node
pub struct WhitelistedTokensProvider(Vec<Address>);

// Implementation used by the EN
#[derive(Debug)]
pub struct RpcWhitelistedTokensProvider(Arc<RwLock<Vec<Address>>>);

// Named task type for convenience.
#[derive(Debug)]
pub struct WhitelistedTokensUpdateTask {
    tokens: Arc<RwLock<Vec<Address>>>,
    main_node_client: _,
}

impl WhitelistedTokensUpdateTask {
    pub fn run(self, stop_receiver: watch::Receiver<bool>) -> anyhow::Result<()> {
        // ...
    }
}
```

Then again, I'm not sure how stable this solution will be (judging by previous discussion, it seems somewhat ad-hoc to me), so maybe it would be an overkill.

Reply:
Yeah, exactly. I decided not to integrate this task into node framework for now. In addition to solution being stable, I'm not sure about the best way to add such "small" tasks to the framework, there are bunch of them running in EN and they are not integrated into framework yet.